### PR TITLE
MQTT for SubmodelService

### PR DESCRIPTION
**Pull request by masud-svg**

### CODE REVIEW

**Review by mdanish98 — COMMENTED** [review]
Please use BaSyx Formatter for all files.

**Review by mdanish98 — CHANGES_REQUESTED** [review]

File: basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/feature/mqtt/MqttSubmodelServiceConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * MQTT configuration to allow for the automatic enable of the feature using
+ * the config file.
+ */
```
[the remark this answers is not in the document]
Comment (masud-svg) [reply basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/feature/mqtt/MqttSubmodelServiceConfiguration.java:45]: updated

File: basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/feature/mqtt/MqttSubmodelServiceTopicFactory.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import org.eclipse.digitaltwin.basyx.common.mqttcore.encoding.Encoder;
+
+/**
+ * MQTT topic factory for the eventing on the submodel level.
+ * 
+ * @author rana
+ */
+public class MqttSubmodelServiceTopicFactory extends AbstractMqttTopicFactory {
+	
+	private static final String SUBMODELSERVIC = "sm-service";
```
[the remark this answers is not in the document]
Comment (masud-svg) [reply basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/feature/mqtt/MqttSubmodelServiceTopicFactory.java]: updated

File: basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/feature/mqtt/MqttSubmodelServiceTopicFactory.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	private static final String SUBMODELS = "submodels";
+	private static final String CREATED = "created";
+	private static final String UPDATED = "updated";
+	private static final String DELETED = "deleted";
+	private static final String SUBMODELELEMENTS = "submodelElements";
```
[the remark this answers is not in the document]
Comment (masud-svg) [reply basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/feature/mqtt/MqttSubmodelServiceTopicFactory.java]: updated

File: basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/feature/mqtt/MqttSubmodelServiceTopicFactory.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+	/**
+	 * @param encoder
```
[the remark this answers is not in the document]
Comment (masud-svg) [reply basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/feature/mqtt/MqttSubmodelServiceTopicFactory.java]: updated

File: basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/feature/mqtt/TestMqttSubmodelObserver.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+import io.moquette.broker.config.ClasspathResourceLoader;
+import io.moquette.broker.config.IConfig;
+import io.moquette.broker.config.IResourceLoader;
+import io.moquette.broker.config.ResourceLoaderConfig;
+
+/**
+ * Tests events for submodelElements in SM Service
+ * @author rana
+ */
+public class TestMqttSubmodelObserver {
```
[the remark this answers is not in the document]
Comment (masud-svg) [reply basyx.submodelservice/basyx.submodelservice-feature-mqtt/src/test/java/org/eclipse/digitaltwin/basyx/submodelrepository/feature/mqtt/TestMqttSubmodelObserver.java:71]: added

**Review by mdanish98 — APPROVED** [review]
Approved

File: basyx.submodelservice/basyx.submodelservice-feature-mqtt/pom.xml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.mqttcore</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.submodelservice-core</artifactId>
+			<classifier>tests</classifier>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
```
Comment (aaronzi) [inline basyx.submodelservice/basyx.submodelservice-feature-mqtt/pom.xml:27]: Duplicate dependency. Please remove

**Review by aaronzi — CHANGES_REQUESTED** [review]

**Pull request merged**